pybaseball/pitching_leaders.py

```diff
@@ -1,81 +1,50 @@
 
-    data = []
-    # pulls headings from the fangraphs table
-    headings = []
-    headingrows = table.find_all('th')
-    for row in headingrows[1:]:
-        headings.append(row.text.strip())
-    
-    data.append(headings)
-    table_body = table.find('tbody')
-    rows = table_body.find_all('tr')
 
-    for row in rows:
-        cols = row.find_all('td')
-        cols = [ele.text.strip() for ele in cols]
-        data.append([ele for ele in cols[1:]])
-
-    data = pd.DataFrame(data)
-    data = data.rename(columns=data.iloc[0])
-    data = data.reindex(data.index.drop(0))
-
-    # replace emptry strings with NaN
-    data.replace(r'^\s*$', np.nan, regex=True, inplace = True)
-
-    # convert all percent strings to proper percetages
-    percentages = ['Contact% (pi)', 'Zone% (pi)','Z-Contact% (pi)','O-Contact% (pi)','Swing% (pi)','Z-Swing% (pi)','O-Swing% (pi)','SL% (pi)','SI% (pi)','SB% (pi)','KN% (pi)','FS% (pi)','FC% (pi)','FA% (pi)','CU% (pi)','CS% (pi)','CH% (pi)','TTO%','Hard%','Med%','Soft%','Oppo%','Cent%','Pull%','K-BB%','Zone% (pfx)','Contact% (pfx)','Z-Contact% (pfx)','O-Contact% (pfx)','Swing% (pfx)','Z-Swing% (pfx)','O-Swing% (pfx)','UN% (pfx)','KN% (pfx)','SC% (pfx)','CH% (pfx)','EP% (pfx)','KC% (pfx)','CU% (pfx)','SL% (pfx)','SI% (pfx)','FO% (pfx)','FS% (pfx)','FC% (pfx)','FT% (pfx)','FA% (pfx)','BB%','K%','SwStr%','F-Strike%','Zone%','Contact%','Z-Contact%','O-Contact%','Swing%','Z-Swing%','O-Swing%','XX%','KN%','SF%','CH%','CB%','CT%','SL%','FB%','BUH%','IFH%','HR/FB','IFFB%','GB%','LD%','LOB%', 'XX% (pi)', 'PO%']
-    for col in percentages:
-        # skip if column is all NA (happens for some of the more obscure stats + in older seasons)
-        if not data[col].empty:
-            if pd.api.types.is_string_dtype(data[col]):
-                data[col] = data[col].astype(str).str.strip(' %')
-                data[col] = data[col].astype(str).str.strip('%')
-                data[col] = data[col].astype(float)/100.
-        else:
-            #print(col)
-            pass
-
-    #convert everything except name and team to numeric
-    cols_to_numeric = [col for col in data.columns if col not in ['Name', 'Team', 'Age Rng', 'Dollars']]
-    data[cols_to_numeric] = data[cols_to_numeric].astype(float)
-
-    #sort by WAR and wins so best players float to the top
-    data = data.sort_values(['WAR', 'W'], ascending=False)
-    #put WAR at the end because it looks better
-    cols = data.columns.tolist()
-    cols.insert(7, cols.pop(cols.index('WAR')))
-    data = data.reindex(columns= cols)
-    return data
-
-def pitching_stats(start_season, end_season=None, league='all', qual=1, ind=1):
+def pitching_stats(start_season: int, end_season: int = None, league: str = 'all', qual: Union[int, str] = 1,
+                   ind: int = 1):
```

Comment:
Re qual, yes, I _know_ this is just the existing thingy, but should we consider changing this?

Reply:
We could treat None = 'y', and then just change it to an Optional[int] with some logic in the function.

Reply:
If you want to do that, I may hold off until my next PR to do so, as I've got something cool in store. :)